src/main/java/edu/kit/provideq/toolbox/api/SubRoutineRouter.java

```
@@ -84,6 +76,63 @@ private Mono<ServerResponse> handleSubRoutineRouteForMetaSolver(MetaSolver<?, ?,
+            .parameter(getParameterBuilder(metaSolver))
+            .tag(problemType.getId())
+            .description("Returns the sub-routines available for the given solver id of type "
+                    + problemType.getId() + ". "
+                    + "Sub-routines are used in some solvers to solve sub-problems. "
+                    + "Passing a sub-routine in a solve request will ensure that"
+                    + " the desired sub-routine is used in the calculation.")
+            .response(responseBuilder()
+                    .responseCode(String.valueOf(HttpStatus.OK.value()))
+                    .content(getOkResponseContent(metaSolver)))
+            .response(responseBuilder()
+                    .responseCode(String.valueOf(HttpStatus.NOT_FOUND.value())));
+  }
+
+  private static org.springdoc.core.fn.builders.parameter.Builder
+      getParameterBuilder(MetaSolver<?, ?, ?> metaSolver) {
+    return parameterBuilder()
+            .in(ParameterIn.QUERY)
+            .name("id")
+            .description("The id of the solver to get the sub-routines from."
+                    + " Use the endpoint GET /solvers/" + metaSolver.getProblemType().getId()
+                    + " to get a list of available solver ids.")
+            .required(true)
+            .example(metaSolver
+                    .getAllSolvers().stream()
+                    .findFirst()
+                    .map(ProblemSolver::getId)
+                    .orElse("Error: no solver found"));
+  }
+
+  private static Builder getOkResponseContent(MetaSolver<?, ?, ?> metaSolver) {
+    String example = metaSolver
+            .getAllSolvers().stream()
+            .findFirst()
+            .map(solver -> {
+              var subRoutines = solver.getSubRoutines();
+              try {
+                return new ObjectMapper().writeValueAsString(subRoutines);
+              } catch (JsonProcessingException e) {
+                return "Error: example could not be parsed";
```

please throw an error instead